### PR TITLE
[Jormun] Fix direct_path_mode with distributed

### PR DESCRIPTION
**Pull request by AurelienLP**
We don't compute pt_journey for modes in `direct_path_mode[]`

### CODE REVIEW

File: source/jormungandr/jormungandr/scenarios/distributed.py [hunk 1]
```diff
@@ -90,8 +90,17 @@ def _compute_all(self, future_manager, request, instance, krakens_call, context)
...
+            mode for mode, _, direct_path_type in krakens_call if direct_path_type != "only"
+        }
+        requested_arr_modes_with_pt = {
+            mode for _, mode, direct_path_type in krakens_call if direct_path_type != "only"
+        }
+
+        requested_direct_path_modes = {
+            mode for mode, _, direct_path_type in krakens_call if direct_path_type == "only"
+        }
+        requested_direct_path_modes.update(requested_dep_modes_with_pt)
```
Comment (xlqian) [inline source/jormungandr/jormungandr/scenarios/distributed.py:106]: a comment on this line will be helpful for further dev

File: source/jormungandr/jormungandr/scenarios/helper_classes/pt_journey.py [hunk 2]
```diff
@@ -214,35 +214,36 @@ def _create_parameters(request):
...
-                arr_mode=arr_mode,
-                periode_extremity=periode_extremity,
-                journey_params=self._journey_params,
-                bike_in_pt=bike_in_pt,
-                request=self._request,
-            )
-
-            self._value.append(PtPoolElement(dep_mode, arr_mode, pt_journey))
+        for dep_mode, arr_mode, dp_type in self._krakens_call:
+            if dp_type != "only":
```
Comment (xlqian) [inline source/jormungandr/jormungandr/scenarios/helper_classes/pt_journey.py]:
```python
if dp_type == "only":
    continue

dp = self._streetnetwork_path_pool.wait_and_get(...
```
you have less indentation to do in this way :)

**Review by xlqian — APPROVED** [review]

File: source/jormungandr/jormungandr/scenarios/distributed.py (hunk 1 above)
Comment (pbougue) [inline source/jormungandr/jormungandr/scenarios/distributed.py:106]: Not sure at all : Aren't you supposed to do that only when requested_direct_path_modes is empty?

File: source/jormungandr/jormungandr/scenarios/helper_classes/pt_journey.py (hunk 2 above)
Comment (pbougue) [inline source/jormungandr/jormungandr/scenarios/helper_classes/pt_journey.py]: Won't you block some kraken calls in new_default?
Not sure this is used there, though.

**Review by pbougue — APPROVED** [review]
Looks OK even if I'm a bit lost in the rationale for that code (trust @xlqian 's review for that).
So the goal is to stop computing PT journeys with modes that are only intended to be direct_paths.
(a bit ambiguous from main comment).

Maybe it's worth launching artemis on this PR.

File: source/jormungandr/jormungandr/scenarios/distributed.py (hunk 1 above)
Comment (AurelienLP) [reply source/jormungandr/jormungandr/scenarios/distributed.py]: > Not sure at all : Aren't you supposed to do that only when requested_direct_path_modes is empty?

We need to compute direct_paths for modes in first_sections_modes AND modes in direct_path_modes

Comment (AurelienLP) [reply source/jormungandr/jormungandr/scenarios/distributed.py:106]: We need to compute direct_paths for modes in first_sections_modes AND modes in direct_path_modes

File: source/jormungandr/jormungandr/scenarios/helper_classes/pt_journey.py (hunk 2 above)
Comment (AurelienLP) [reply source/jormungandr/jormungandr/scenarios/helper_classes/pt_journey.py]: Nice catch !

Comment (AurelienLP) [reply source/jormungandr/jormungandr/scenarios/helper_classes/pt_journey.py]: This part is distributed only

**Review by xlqian — APPROVED** [review]

**Comment by xlqian** [comment]
References of artemis may need to be updated.

**Review by pbougue — APPROVED** [review]
Thanks for clear comments :+1:

**Comment by AurelienLP** [comment]
Jenkins test this please

**Comment by AurelienLP** [comment]
Jenkins, test this please

**Pull request merged**